File: Javascript/InsertionSort.js
const prompt=require("prompt-sync")({sigint:true});
function Insertionsort(arr,size){
    for(let i=1;i<size;i++){
        let min=arr[i];
        let j=i-1;
        while(j>=0 && arr[j]>min){
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=min;
    }
    console.log(arr);
}
var size=prompt("Enter Array of Size => ");
var arr=[];
for(let i=0;i<size;i++){
arr[i]=parseInt(prompt());
}
Insertionsort(arr,size);

File: Go/Insertionsort.go
package main

import (
	"fmt"
)

func Insertionsort(arr []int) {
	var i, k, j int
	for i = 0; i < len(arr); i++ {
		k = arr[i]
		j = i - 1
		for j >= 0 && arr[j] > k {
			arr[j+1] = arr[j]
			j = j - 1
		}
		arr[j+1] = k
	}
	fmt.Print(arr)
}
func main() {
	var i, j int
	fmt.Print("Enter The Size of Array => ")
	fmt.Scan(&i)
	arr := make([]int, i)
	for j = 0; j < i; j++ {
		fmt.Scan(&arr[j])
	}
	Insertionsort(arr)

}
